### PR TITLE
Add in a tab for in-app "lite" image editing (e.g., cropping, masking, resampling, etc.) using locally cached version of https://www.photopea.com

### PR DESCRIPTION
**Pull request by ngvtuan**
##### Commits
- [x] Add in a tab for in-app "lite" image editing (e.g., cropping, masking, resampling, etc.) using locally cached version of https://www.photopea.com
- [x] Add loading component

##### FreedCamp
- [x] [Add in a tab for in-app "lite" image editing (e.g., cropping, masking, resampling, etc.) using locally cached version of https://www.photopea.com] (https://freedcamp.com/dashboard/home/panel/task/38401307)

### CODE REVIEW

File: src/features/pastelLoading/PastelLoading.module.css [hunk 1]
```diff
@@ -0,0 +1,81 @@
+.loading {
```
Comment (br4in3x) [inline src/features/pastelLoading/PastelLoading.module.css:1]: I think we can get away by using one of loaders from http://jonjaques.github.io/react-loaders/ - the "ball beat" type looks the closest to what you made. Reasons to not have our own custom loader for now is:
1. It's generic - doesn't specified in our design and can be anything
2. Increases complexity - we don't need to write a generic loader if there is an established library of generic loaders with 0 dependencies

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        >
+          X
+        </button>
+        <div className={styles.iframe}>
+          <PastelLoading loadingIsOpen={isLoading} />
+          <iframe
+            id='photopea'
+            src='https://www.photopea.com'
+            onLoad={() => setTimeout(() => setIsLoading(false), 100)}
+          />
```
Comment (br4in3x) [inline src/features/pastelPhotopea/PastelPhotopeaModal.tsx:38]: Can we have this preloaded? It takes significant time to show up. Here is an example:
https://codepen.io/darknessHO/pen/goNvdK. I think we can use [React Portals](https://reactjs.org/docs/portals.html) for that. We can have this iframe load and be invisible and render it in this component through a portal. 

This is just an idea, perhaps you can come up with your own mechanism :)

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          onClick={() => dispatch(closePastelPhotopeaModal())}
+        >
+          X
+        </button>
+        <div className={styles.iframe}>
+          <PastelLoading loadingIsOpen={isLoading} />
+          <iframe
+            id='photopea'
+            src='https://www.photopea.com'
+            onLoad={() => setTimeout(() => setIsLoading(false), 100)}
```
Comment (br4in3x) [inline src/features/pastelPhotopea/PastelPhotopeaModal.tsx:37]: Why do we need `setTimeout` here?

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import styles from './PastelPhotopeaModal.module.css'
+import { closePastelPhotopeaModal } from './PastelPhotopeaModalSlice'
+
+export default function PastelPhotopeaModal(): JSX.Element {
+  const { modalIsOpen } = useAppSelector(state => state.pastelPhotopeaModal)
+  const dispatch = useAppDispatch()
+  const [isLoading, setIsLoading] = useState(true)
+
+  if (!modalIsOpen) {
+    return <></>
```
Comment (br4in3x) [inline src/features/pastelPhotopea/PastelPhotopeaModal.tsx:15]: Perhaps `return null` is better?

**Review by br4in3x — CHANGES_REQUESTED** [review]

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  return (
+    <Modal
```
Comment (br4in3x) [inline src/features/pastelPhotopea/PastelPhotopeaModal.tsx:19]: When the modal opens, it has a large white border. Let's remove the border to provide more space for the iframe.

File: src/features/pastelLoading/PastelLoading.module.css (hunk 1 above)
Comment (ngvtuan) [reply src/features/pastelLoading/PastelLoading.module.css]: @br4in3x I have updaed for the loading by removed it

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx (hunk 2 above)
Comment (ngvtuan) [reply src/features/pastelPhotopea/PastelPhotopeaModal.tsx]: yes we can i have updated the code

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx (hunk 4 above)
Comment (ngvtuan) [reply src/features/pastelPhotopea/PastelPhotopeaModal.tsx]: Updated

File: src/features/pastelPhotopea/PastelPhotopeaModal.tsx (hunk 5 above)
Comment (ngvtuan) [reply src/features/pastelPhotopea/PastelPhotopeaModal.tsx:13]: Updated

**Review by br4in3x — APPROVED** [review]

**Review by br4in3x — APPROVED** [review]

**Pull request merged**